bpfd/src/bpf.rs

```diff
@@ -105,27 +55,11 @@ impl DispatcherProgram {
     }
 }
 
-#[derive(Debug, Clone)]
-pub(crate) struct InterfaceInfo {
-    pub(crate) xdp_mode: String,
-    pub(crate) programs: Vec<ProgramInfo>,
-}
-
-#[derive(Debug, Clone)]
-pub(crate) struct ProgramInfo {
-    pub(crate) id: String,
-    pub(crate) name: String,
-    pub(crate) path: String,
-    pub(crate) position: usize,
-    pub(crate) priority: i32,
-    pub(crate) proceed_on: Vec<i32>,
-}
-
 pub(crate) struct BpfManager<'a> {
     config: &'a Config,
     dispatcher_bytes: &'a [u8],
     dispatchers: HashMap<u32, DispatcherProgram>,
-    programs: HashMap<u32, HashMap<Uuid, ExtensionProgram>>,
+    programs: HashMap<Uuid, Program>,
```

Comment:
So, it looks like you just have one list of programs indexed by UUID.

Reply:
Yes that's correct.